### ISSUE TITLE
SNMPv3 input and secretstores support

### ISSUE THREAD

**Issue by KoMa1978** [influxdata/telegraf#14972]
### Use Case

It would be great if the SNMP plugin supports secretstores for AUTH and PRIV passwords.

### Expected behavior

Do not have to enter passwords in config file directly.

### Actual behavior

Is not supported

### Additional info

Telegraf version 1.29.5

**Comment by srebhan** [comment]
@KoMa1978 can you please test the binary in #14975 available as-soon-as CI finishes the tests successfully?! Let me know if this is what you intended!

**Comment by maintain3r** [comment]
As a secret store i rely on docker secrets, but would be interested in seeing other options.

**Comment by srebhan** [comment]
@maintain3r sorry I don't get what you are requesting... :-/ The corresponding PR allows to use secrets for the two configuration fields of SNMP-family plugins. You can then use those options with available secret-stores...

Are you requesting to convert more fields of SNMP to secrets? Or are you looking for an additional type of secret store? Or something else?

**Comment by maintain3r** [comment]
Sorry if it wasn't clear, all Im saying is that years ago I was also trying to figure out a best way how to hide secrets for telegraf snmp plugin. Ended up with sotring them in dockers secrets which allows to mount them into my modified telegraf container. In the telegraf config file I reference the var names which are env vars that I mount as part of my bootstrap script.
In the context of this topic I don;t necessarily request a feature, I rather wanted to see what are the options to have secrets hidden in my telegraf config file. I didn;t understand what exactly the approach and the request of this topic is, so that's the reason of asking questions, like maybe there's abetter way of doing what I need.
Hope it makes sense.

**Comment by KoMa1978** [comment]
@maintain3r you can now use the secretstores functionality of telegraf also in the snmp input plugin for the auth and priv passwords. Working fine for me during the first tests.

**Comment by srebhan** [comment]
@KoMa1978 the PR will be merged into the master branch and probably released with v1.31.0 due in June.

**Comment by KoMa1978** [comment]
Again @srebhan thanks a lot! Feel free to close this issue. I think it is solved and I'm looking forward for the next release.

**Comment by srebhan** [comment]
@KoMa1978 please leave the issue open, it will be closed automatically once the PR is merged...
